src/main/java/io/cdap/plugin/salesforce/authenticator/Authenticator.java

```diff
@@ -84,6 +85,10 @@ public static OAuthInfo getOAuthInfo(AuthenticatorCredentials credentials) throw
       return oAuthInfo;
     }
 
+    if (credentials.getGrantType() == null) {
```

Comment:
No, we are populating grant type based on the set of parameters being provided by the user

Reply:
wouldn't it be confusing for user to enter only partial details, it would be better from UI perspective if we can hide the fields which are not needed for a specific grant type.

Reply:
The Authenticator class fields are directly exposed to the UI? I thought that was only via the plugin configs

Reply:
I didn't add the field to all the plugin configs because I was unsure if the grant type itself is something that we expose to the user. But if it is required, we can change all the configs too to have this additional field.

Reply:
its better to do it. @MrRahulSharma you can confirm. Salesforce has documented these grant types in detail.

Reply:
The plan currently is not to show this to CDF users, and only have the grant implementation for internal usage. As AFAIR there are no customer ask requiring this grant in CDF.